api/bucket.go

```diff
@@ -123,6 +123,17 @@ func ListBuckets(c *gin.Context) {
 
 // ListManagedBuckets lists managed buckets for the user when no secret is provided
 func ListManagedBuckets(c *gin.Context) {
+
+	type secretData struct {
+		SecretId   string `json:"secretId"`
+		SecretName string `json:"secretName"`
+	}
+	type bucketWithSecretName struct {
+		*objectstore.BucketInfo
+		SecretRef  string      `json:"secretId,-,omitempty"`
+		SecretInfo *secretData `json:"secretInfo"`
```

Comment:
The `secretId` in `secretData` struct seems to be redundant as that field already available through `BucketInfo`.

Can't we just do
```
type bucketWithSecretName struct {
  *objectstore.BucketInfo
  SecretName string `json:"secretName,omitempty"`
}
```

Reply:
i have slightly changed this - these json binding tricks are only for returning nicer jsons to callers (json response is enhanced with data that isn;t required in the "api" or "domain" structs